receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -130,20 +131,40 @@ func (r *pReceiver) startTargetAllocator(allocConf *targetAllocator, baseCfg *co
 	return nil
 }
 
+// Calculate a hash for a scrape config map.
+// This is done by marshaling to YAML because it's the most straightforward and doesn't run into problems with unexported fields.
+func getScrapeConfigHash(jobToScrapeConfig map[string]*config.ScrapeConfig) (hash.Hash64, error) {
+	var err error
+	hash := fnv.New64()
+	yamlEncoder := yaml.NewEncoder(hash)
+	for jobName, scrapeConfig := range jobToScrapeConfig {
```

Comment:
> The iteration order over maps is not specified and is not guaranteed to be the same from one iteration to the next.

- [Go spec](https://go.dev/ref/spec#For_statements)

I don't think this is going to be a viable approach.  To ensure consistency I believe it necessary to first extract the keys from the map into a slice and sort them, then iterate over the sorted slice.

Reply:
@Aneurysm9 - I did find another way to fix this issue using a different package. Please lmk if you see any concerns.

Reply:
@Aneurysm9 - here are the results for the benchmark ( see the test for the benchmark tests i used. This is just for reference and I can remove them after you review)
![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/36861737/00e3da53-3c98-40c4-b04b-12adcb0deee6)

![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/36861737/37299846-87e2-4ef6-8d51-c48dfeeee0b2)

The new package structhash seems to have better performance

Reply:
Hi @Aneurysm9, Following up. Please lmk if you have any suggestions/feedback. Thanks for looking into this.

Reply:
@Aneurysm9 - Following up. Please lmk if you have any concerns/feedback. Thanks!